api.go

```diff
@@ -122,6 +122,42 @@ func NewApiWithClient(a ma.Multiaddr, c *gohttp.Client) (*HttpApi, error) {
 	return api, nil
 }
 
+// NewDirectAPIClient is used to instantiate a HttpApi client
+// that connects to an endpoint which leverages additional http paths.
+//
+// If you need to connect to a IPFS HTTP API located at https://foo.bar/baz/api/v0
+// you should use NewDirectAPIClient.
+func NewDirectAPIClient(url string) (*HttpApi, error) {
+	api := &HttpApi{
+		url: url,
+		httpcli: gohttp.Client{
+			Transport: &gohttp.Transport{
+				Proxy:             gohttp.ProxyFromEnvironment,
+				DisableKeepAlives: true,
+			},
+		},
+		applyGlobal: func(*RequestBuilder) {},
+	}
+
+	// We don't support redirects.
+	api.httpcli.CheckRedirect = func(_ *gohttp.Request, _ []*gohttp.Request) error {
+		return fmt.Errorf("unexpected redirect")
+	}
+
+	return api, nil
+}
+
+// WithAuthorization is used to wrap an instance of HttpApi
+// with an authenticated transport, such as JWT
+func (api *HttpApi) WithAuthorization(header, value string) *HttpApi {
```

Comment:
I'm worried users will want to specify multiple headers, cookies, and a bunch of other random tweaks. See my comment above about dropping this.

It might also make sense to have `SetCookie`, `SetAuthorization`, and `SetHeader` functions but I'd prefer to go with the simplest option.

Reply:
Ah yes that's a very good point, It might make sense to just have a generalized `SetHeader` instead since ultimately that can be used to manipulate all cookies, and header tweaks. Perhaps something like

```Golang

func (api *HttpApi) SetHeader(headers map[string][]string) (*HttpApi, error) {
        if len(headers) == 0 {
              return errors.New("headers can't be nil")
        }
	return &HttpApi{
		url: api.url,
		httpcli: gohttp.Client{
			Transport: newAuthenticatedTransport(api.httpcli.Transport, headers),
		},
	}
}

type transport struct {
	headers map[string][]string
	httptr        http.RoundTripper
}

func newAuthenticatedTransport(tr http.RoundTripper, headers map[string][]string) *transport {
	return &transport{
                headers: headers,
		httptr: tr,
	}
}

func (t *transport) RoundTrip(req *http.Request) (*http.Response, error) {
      // generate the round tripper
      for header, values := range t.headers {
            for _, value := range values {
                req.Header.Add(header, value)
            }
        }
	return t.httptr.RoundTrip(req)
}

Reply:
I'm wondering if we just want to expose a `Header() http.Header` method like in `ResponseWriter`. Alternatively, we could change the HttpApi type signature to:

```go
type HttpApi struct {
  Client http.Client
  Header http.Header
  // private fields...
}
```

@magik6k, do you have any opinions on this?

Reply:
:+1:  that sounds even better.

Reply:
That works for me, there isn't much reason to hide those